### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by LaskoA**
Please check my solution, thank you

.

### CODE REVIEW

File: taxi/forms.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        license_number = self.cleaned_data["license_number"]
+        if len(license_number) != 8:
+            raise ValidationError("License number should have 8 symbols")
+        elif not license_number[0:3].isalpha():
+            raise ValidationError("First 3 symbols should be letters")
+        elif not license_number[0:3].isupper():
+            raise ValidationError("First 3 letters should be uppercase")
+        elif not license_number[3:].isnumeric():
+            raise ValidationError("Last 5 symbols should be digits")
+        return license_number
```
Comment (Abnormaltype) [inline taxi/forms.py:23]: DRY) fix it)

File: taxi/urls.py [hunk 2]
```diff
@@ -1,14 +1,70 @@
 from django.urls import path
 
-from .views import index, CarListView, CarDetailView, DriverListView, DriverDetailView, ManufacturerListView
+from .views import (index,
```
Comment (Abnormaltype) [inline taxi/urls.py:3]: fix indentation

File: templates/taxi/car_form.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    <h1>{{ object|yesno:"Update,Create" }} Car</h1>
+
+    <form action="" method="post" novalidate>
+        {% csrf_token %}
+        {{ form|crispy }}
+
+        <input class="btn btn-primary" type="submit" value="Submit">
+
+    </form>
+{% endblock %}
```
Comment (Abnormaltype) [inline templates/taxi/car_form.html:15]: add blank for the end of the file

File: templates/taxi/driver-license-update.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+    <h1>Update license</h1>
+    <br>
+    <p>You are about to update license <strong>{{ driver.license_number }}</strong></p>
+    <form action="" method="post" novalidate>
+    {% csrf_token %}
+        {{ form|crispy }}
+        <input type="submit" value="Update" class="btn btn-danger" />
+    </form>
+{% endblock %}
```
Comment (Abnormaltype) [inline templates/taxi/driver-license-update.html:14]: the same here

File: templates/taxi/driver_form.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+{% block content %}
+    <h1>Create Driver</h1>
+    <br>
+    <form action="" method="post" novalidate>
+    {% csrf_token %}
+        {{ form|crispy }}
+    <input type="submit" value="Submit">
+    </form>
+{% endblock %}
```
Comment (Abnormaltype) [inline templates/taxi/driver_form.html:12]: the same here

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Comment by LaskoA** [comment]
Please check again after my fixes, thank you)

File: taxi/urls.py [hunk 6]
```diff
@@ -1,14 +1,71 @@
...
+                DriverListView,
+                DriverDetailView,
+                DriverCreateView,
+                DriverDeleteView,
+                DriverLicenseUpdateView,
+                ManufacturerListView,
+                ManufacturerCreateView,
+                ManufacturerUpdateView,
+                ManufacturerDeleteView,
+                )
```
Comment (Abnormaltype) [inline taxi/urls.py:20]:
```suggestion
    index,
    assign_driver,
    CarListView,
    CarDetailView,
    CarCreateView,
    CarUpdateView,
    CarDeleteView,
    DriverListView,
    DriverDetailView,
    DriverCreateView,
    DriverDeleteView,
    DriverLicenseUpdateView,
    ManufacturerListView,
    ManufacturerCreateView,
    ManufacturerUpdateView,
    ManufacturerDeleteView,
)
```

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Comment by LaskoA** [comment]
Attempt #3, thank you)

**Review by Abnormaltype — APPROVED** [review]
Well done!

**Pull request closed without merging**